tests/functional/src/main/java/org/mule/functional/junit4/runners/ArtifactClassloaderRunnerContextAware.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.functional.junit4.runners;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Defines that the test is going to be run with an {@link ArtifactClassloaderTestRunner} and would need
```

Comment:
The name is a misleading to me, the class won't be aware of the ArtifactClassloaderTestRunner, it will be just aware of the plugin classLoaders